src/DebugConfigurationProvider.ts

```diff
@@ -0,0 +1,86 @@
+import * as vscode from 'vscode'
+import { getTestCommand, isCRATestCommand } from './helpers'
+
+export class DebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+  private fileNameToRun: string = ''
+  private testToRun: string = ''
+
+  /**
+   * Prepares injecting the name of the test, which has to be debugged, into the `DebugConfiguration`,
+   * This function has to be called before `vscode.debug.startDebugging`.
+   */
+  public prepareTestRun(fileNameToRun: string, testToRun: string) {
+    this.fileNameToRun = fileNameToRun
+    this.testToRun = testToRun
+  }
+
+  resolveDebugConfiguration(
+    _folder: vscode.WorkspaceFolder | undefined,
+    debugConfiguration: vscode.DebugConfiguration,
+    _token?: vscode.CancellationToken
+  ) {
+    if (debugConfiguration.name !== 'vscode-jest-tests') {
+      return debugConfiguration
+    }
+
+    if (!debugConfiguration.env) {
+      debugConfiguration.env = {}
+    }
+    // necessary for running CRA test scripts in non-watch mode
+    debugConfiguration.env.CI = 'vscode-jest-tests'
+
+    if (!debugConfiguration.args) {
+      debugConfiguration.args = []
+    }
+    if (this.fileNameToRun) {
+      debugConfiguration.args.push(this.fileNameToRun)
+      if (this.testToRun) {
+        debugConfiguration.args.push('--testNamePattern')
+        debugConfiguration.args.push(this.testToRun)
+      }
+      this.fileNameToRun = ''
+      this.testToRun = ''
+    }
+
+    return debugConfiguration
+  }
+
+  provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, _token?: vscode.CancellationToken) {
+    // default jest config according to:
+    // https://github.com/Microsoft/vscode-recipes/tree/master/debugging-jest-tests#configure-launchjson-file-for-your-test-framework
+
+    // create-react-app config according to:
+    // https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#debugging-tests-in-visual-studio-code
+
+    const debugConfiguration: vscode.DebugConfiguration = {
+      type: 'node',
+      name: 'vscode-jest-tests',
+      request: 'launch',
+      args: ['--runInBand'],
+      cwd: '${workspaceFolder}',
+      console: 'integratedTerminal',
+      internalConsoleOptions: 'neverOpen',
+    }
+
+    const testCommand = folder && getTestCommand(folder.uri.fsPath)
+    if (isCRATestCommand(testCommand)) {
+      const craCommand = testCommand.split(' ')
+      // Settings specific for projects bootstrapped with `create-react-app`
+      debugConfiguration.runtimeExecutable = '${workspaceFolder}/node_modules/.bin/' + craCommand.shift()
+      debugConfiguration.args = [...craCommand, ...debugConfiguration.args]
+      debugConfiguration.protocol = 'inspector'
+    } else if (testCommand === 'node scripts/test.js --env=jsdom') {
```

Comment:
me: _"uh-oh, this is fragile..."_

me: _"yeah, but that is ok because if it failed developers can just override it in their custom debug config"_

me: _"ok, then why do we even need this code here? when developers eject CRA, I think they would expect some adjustment to their environment, such as their debug config; on the other hand, it is not so intuitive that their trusted vscode-jest suddenly failed to debug after they changed the test script in package.json."_

me: _"maybe, but if vscode-jest can work out-of-box in their CRA environment, wouldn't it be "intuitive" to think it should work in ejected env as well? if the testCommand check is too fragile, there are many ways to improve it: we can parse all package.json scripts,  parse the testCommand further, go down to the file system to find the binary, just to name a few; and if different CRA version has different test scripts, not a problem, if we can find CRA binary, we can find its version... there is no problem we can't resolve!"_

me: _"This is what we call the slippery slope... at the beginning, everything looked equally plausible, but as you travel down the path, the effect starts to compound...before you know it, there will be hundreds of lines of code serving CRA variations and yet still insufficient... now just imagine if we have chosen the other path..."_

----

 "Just because we can doesn't mean we should" 😏

Reply:
> Should I remove it?

sounds good to me.

Reply:
Is removed.